### PR TITLE
Accomodate for bm ports on cleanup

### PR DESCRIPTION
**Pull request by sebageek**
Not tested in qa yet.

### CODE REVIEW

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py [hunk 1]
```diff
@@ -210,6 +210,10 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
 
         for other_binding in other_bindings:
             _, other_binding_host_config = self._host_or_host_group(other_binding.host)
+            if not other_binding_host_config:
+                LOG.warning("No config available / invalid binding host %s, segment %s with profile %s in network %s",
+                            other_binding.host, other_binding.segment_id, other_binding.profile, network['id'])
```
Comment (joker-at-work) [inline networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: Are you sure, `other_binding` alwas has all those attributes?

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py [hunk 2]
```diff
@@ -203,12 +204,28 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
...
+            host = other_binding.host
+
+            # try to get alternate host from binding profile
+            binding_profile = None
+            try:
+                binding_profile = json.loads(other_binding.profile)
+                if binding_profile:
+                    switch = CiscoACIMechanismDriver.switch_from_local_link(binding_profile)
+                    if switch:
+                        host = host
```
Comment (joker-at-work) [inline networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: is this a NOOP?

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py [hunk 3]
```diff
@@ -203,12 +204,28 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
...
+            except json.decoder.JSONDecodeError:
+                # don't act on broken binding profile
+                pass
+
             _, other_binding_host_config = self._host_or_host_group(other_binding.host)
```
Comment (joker-at-work) [inline networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: Do you need to use `host` here, too?

**Review by joker-at-work — CHANGES_REQUESTED** [review]
commit-message:
> it checks if are any
+ there

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py (hunk 2 above)
Comment (sebageek) [reply networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: Yes, I meant to use switch.

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py (hunk 3 above)
Comment (sebageek) [reply networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: Yes. The `host = host` apparently prevented flake8 from warning me.

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py (hunk 1 above)
Comment (sebageek) [reply networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: Portbinding.profile was missing from the SQL query, fixed that. All attributes should now be present as a result of the query, or did you see anything else?

Comment (joker-at-work) [reply networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: I just wasn't sure about the model at all. Was trying to see if you checked it :)

File: networking_aci/plugins/ml2/drivers/mech_aci/driver.py [hunk 4]
```diff
@@ -200,16 +201,37 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
...
+                    if switch:
+                        host = switch
+            except json.decoder.JSONDecodeError:
+                # don't act on broken binding profile
+                pass
+
+            _, other_binding_host_config = self._host_or_host_group(host)
+            if not other_binding_host_config:
+                LOG.warning("No config available / invalid binding host %s, segment %s with profile %s in network %s",
+                            other_binding.host, other_binding.segment_id, other_binding.profile, network['id'])
```
Comment (joker-at-work) [inline networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: This log message needs to use `host` instead of `other_binding.host`, too, doesn't it?

Comment (sebageek) [reply networking_aci/plugins/ml2/drivers/mech_aci/driver.py]: I originally intended here to show binding host, segemnt id and binding profile from the database to show to the user on what basis the driver decided to print this message. Using host would also be an option, then we would only show the derived data... Hmm. Doesn't sound like a bad idea.

**Pull request closed without merging**